tests/forte/processors/remote_processor_test.py

```diff
@@ -111,12 +119,23 @@ def test_remote_processor(self, input_output_pair):
 
         # Configure RemoteProcessor into test mode
         remote_processor: RemoteProcessor = RemoteProcessor()
-        remote_processor.set_test_mode(serve_pl._remote_service_app)
+        remote_processor.set_test_mode(
+            serve_pl._remote_service_app(name=service_name)
+        )
 
         # Build test pipeline
         test_pl: Pipeline[DataPack] = Pipeline[DataPack]()
         test_pl.set_reader(StringReader())
-        test_pl.add(remote_processor)
+        test_pl.add(
+            remote_processor,
+            config={
+                "do_validation": True,
+                "expected_name": service_name,
+                "expected_records": json.dumps(
```

Comment:
why there is a `json.dumps` here? this is a user-oriented interface, we should avoid these extra operations to be done by the caller.